R/serializer.R

```diff
@@ -558,68 +573,142 @@ serializer_device <- function(type, dev_on, dev_off = grDevices::dev.off) {
       } else {
         success(result)
       }
-    }
+    },
+    serializer_params = serializer_params
   )
 }
 
+serializer_param_width <- function() {
+  list(plot_width = list(desc="Width of plot image in units", type="number", required=FALSE, isArray=FALSE))
+}
+serializer_param_height <- function() {
+  list(plot_height = list(desc="Height of plot image in units", type="number", required=FALSE, isArray=FALSE))
+}
+serializer_param_res <- function() {
+  list(plot_res = list(desc="Resolution of plot image", type="number", required=FALSE, isArray=FALSE))
+}
+serializer_param_units <- function() {
+  list(plot_units = list(desc="Units of plot image", type="string", required=FALSE, isArray=FALSE))
+}
+serializer_param_pointsize <- function() {
+  list(plot_pointsize = list(desc="Point size of plot image - TODO - better desc", type="number", required=FALSE, isArray=FALSE))
+}
+serializer_param_bg <- function() {
+  list(plot_bg = list(desc="Background colour of plot image", type="string", required=FALSE, isArray=FALSE))
+}
+
+serializer_param_list <- function(all_except = character()) {
+  params <- c(
+    serializer_param_width(),
+    serializer_param_height(),
+    serializer_param_res(),
+    serializer_param_units(),
+    serializer_param_pointsize(),
+    serializer_param_bg()
+  )
+
+  # oh I wish I had purrr!
+  ignored <- Map(function(x) {
+    params[[x]] <<- NULL
+  }, all_except)
+
+  params
+}
+
+
+serialize_dimensions_args_preparer <- function(req, ...) {
```

Comment:
Matching cosmetic change below

```suggestion
serialize_dev_args_preparer <- function(req, ...) {
```